.github/workflows/ci.yml

```
@@ -0,0 +1,105 @@
+name: Python CI
+
+on:
+  push:
+    branches:
+      - master
+    tags: ['*']
+  pull_request:
+
+concurrency:
+  group: test-${{ github.head_ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHONUNBUFFERED: "1"
+  FORCE_COLOR: "1"
+
+jobs:
+  build:
+    name: Python ${{ matrix.python-version }} ${{ matrix.extra-env }}
+    if: >-
+        !contains(github.event.head_commit.message, '[skip ci]')
+        && !contains(github.event.head_commit.message, '[skip tests]')
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+        include:
+          - python-version: "3.7"
+          - python-version: "3.7"
+            extra-env: "PIP_EXTRA_INSTALLATION='numpy symengine==0.5.0' USE_SYMENGINE=1"
+          - python-version: "3.7"
+            extra-env: "PIP_EXTRA_INSTALLATION='sympy==1.4'"
+          - python-version: "3.8"
+            extra-env: "PIP_EXTRA_INSTALLATION='https://github.com/sympy/sympy/archive/master.zip'"
```

Are you sure this shouldn't be
```suggestion
          - python-version: "3.8"
            env: 
              PIP_EXTRA_INSTALLATION: 'https://github.com/sympy/sympy/archive/master.zip'
```

---

`extra-env` seems to have no effect on the environment variables after searching in github docs. If the suggested update is adopted, other places mentioning it should be updated as well, such as in `jobs.build.name`.

---

One more thing to notice is that, the background of these lines is that 3.7 was the stable version of Python, and we needed sympy master to run 3.8. This is no longer the case, so the matrix should be simplified to just run for Python 3.8 and above (sympy has dropped support for Python 3.7). No need to tend to these details now, maybe just use plain `pip install sympy` with no versions specified for all versions and an extra master build for the latest python version like 3.11 today.